app/src/main/java/com/framgia/moviedbtraining/MainActivity.kt

```diff
@@ -156,8 +158,15 @@ class MainActivity : AppCompatActivity() {
   override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
     super.onActivityResult(requestCode, resultCode, data)
     if (resultCode == Activity.RESULT_OK) {
-      val user: User = mPref!!.getUser()
-      loadUserDetails(user)
+      if (requestCode == LOGIN_INTENT) {
+        val user: User = mPref!!.getUser()
+        loadUserDetails(user)
+        return
+      }
+      if (requestCode == PROFILE_INTENT) {
+        setDataProfile()
+      }
+
```

Comment:
I think you can use  the (switch-case)structure at here for clearly (in the kotlin language is when structure )